File: solutions/leetcode/convert-an-array-into-a-2d-array-with-conditions/main.py
class Solution:

    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        num_freq = defaultdict(int)
        result = []
        for v in nums:
            row = num_freq[v]
            if len(result) == row:
                result.append([])
            result[row].append(v)
            num_freq[v] += 1
        return result

File: solutions/leetcode/convert-an-array-into-a-2d-array-with-conditions/main.cpp
class Solution {
   public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        unordered_map<int, int> numFreq;
        vector<vector<int>> result;
        for (int v : nums) {
            int row = numFreq[v]++;
            if (result.size() == row) {
                result.push_back({});
            }
            result[row].push_back(v);
        }
        return result;
    }
};
